micrometer-meter-provider/src/integrationTest/java/io/opentelemetry/contrib/metrics/micrometer/PrometheusIntegrationTest.java

```diff
@@ -0,0 +1,406 @@
+        .contains("longHistogram_units_count{key1=\"value2\",} 1.0")
+        .contains("longHistogram_units_sum{key1=\"value2\",} 2.0")
+        .contains("# HELP longHistogram_units_max LongHistogram test")
+        .contains("# TYPE longHistogram_units_max gauge")
+        .contains("longHistogram_units_max{key1=\"value1\",} 1.0")
+        .contains("longHistogram_units_max{key1=\"value2\",} 2.0");
+
+    longHistogram.record(2, Attributes.of(KEY1, "value1"));
+
+    output = prometheusMeterRegistry.scrape();
+    assertThat(output)
+        .contains("longHistogram_units_count{key1=\"value1\",} 2.0")
+        .contains("longHistogram_units_sum{key1=\"value1\",} 3.0")
+        .contains("longHistogram_units_max{key1=\"value1\",} 2.0");
+  }
+
+  @Test
+  void observableDoubleGauge() {
+    try (ObservableDoubleGauge observableDoubleGauge =
+        meter
+            .gaugeBuilder("doubleGauge")
+            .setDescription("DoubleGauge test")
+            .setUnit("units")
+            .buildWithCallback(
+                measurement -> {
+                  measurement.record(1.5, Attributes.of(KEY1, "value1"));
+                  measurement.record(2.5, Attributes.of(KEY1, "value2"));
+                })) {
+
+      String output = scrapeFor("doubleGauge");
+      assertThat(output)
+          .contains("# HELP otel_polling_meter")
+          .contains("# TYPE otel_polling_meter untyped")
+          .contains("# TYPE doubleGauge_units gauge")
+          .contains("# HELP doubleGauge_units DoubleGauge test")
+          .contains("doubleGauge_units{key1=\"value1\",} 1.5")
+          .contains("doubleGauge_units{key1=\"value2\",} 2.5");
+    }
+  }
+
+  @Test
+  void observableLongGauge() {
+    try (ObservableLongGauge observableLongGauge =
+        meter
+            .gaugeBuilder("longGauge")
+            .ofLongs()
+            .setDescription("LongGauge test")
+            .setUnit("units")
+            .buildWithCallback(
+                measurement -> {
+                  measurement.record(1, Attributes.of(KEY1, "value1"));
+                  measurement.record(2, Attributes.of(KEY1, "value2"));
+                })) {
+
+      String output = scrapeFor("longGauge");
+      assertThat(output)
+          .contains("# HELP otel_polling_meter")
+          .contains("# TYPE otel_polling_meter untyped")
+          .contains("# TYPE longGauge_units gauge")
+          .contains("# HELP longGauge_units LongGauge test")
+          .contains("longGauge_units{key1=\"value1\",} 1.0")
+          .contains("longGauge_units{key1=\"value2\",} 2.0");
+    }
+  }
+
+  private String scrapeFor(String value) {
+    String output = prometheusMeterRegistry.scrape();
+    if (!output.contains(value)) {
+      output = prometheusMeterRegistry.scrape();
+    }
+    assertThat(output).contains(value);
+    return output;
+  }
+
+  private <T> Consumer<T> onlyOnce(Consumer<T> consumer) {
+    return new Consumer<T>() {
+      boolean first = true;
```

Comment:
It'd probably be safer to use `AtomicBoolean`, just in case (to prevent test flakiness)